### PR TITLE
Minor cleanups

### PR DESCRIPTION
**Pull request by partouf**

### CODE REVIEW

**Comment by partouf** [comment]
Also want to get rid of the editorId reference in the tool-input-view, trying to figure out if it's needed for something.

**Comment by mattgodbolt** [comment]
> Also want to get rid of the editorId reference in the tool-input-view, trying to figure out if it's needed for something.

I'm guessing probably not, it's likely a copy pasta

File: static/panes/compiler.js [hunk 1]
```diff
@@ -1547,10 +1546,16 @@ Compiler.prototype.updateCompilerUI = function () {
...
+
+    this.deferCompiles = true;
+    this.needsCompile = true;
+
     this.updateLibraries();
     this.saveState();
-    this.compile();
     this.updateCompilerUI();
+
+    this.undefer();
```
Comment (mattgodbolt) [inline static/panes/compiler.js:1557]: Oh nice! This looks like an actual bug fix!

File: static/panes/tool.js [hunk 2]
```diff
@@ -215,16 +215,16 @@ Tool.prototype.getInputArgs = function () {
...
         this.monacoStdinField = input;
         this.onOptionsChange();
         this.eventHub.emit('toolSettingsChange', this.compilerId);
     }
 };
 
-Tool.prototype.onToolInputViewClosed = function (compilerId, editorId, toolId, input) {
-    if (this.compilerId === compilerId && this.editorId === editorId && this.toolId === toolId) {
+Tool.prototype.onToolInputViewClosed = function (compilerId, toolId, input) {
+    if (this.compilerId === compilerId && this.toolId === toolId) {
```
Comment (mattgodbolt) [inline static/panes/tool.js:227]: This, and the onToolInputChange is the only thing that had me thinking. But I've convinced myself it's OK :)

**Review by mattgodbolt — APPROVED** [review]
LGTM!

File: static/panes/tool.js (hunk 2 above)
Comment (partouf) [reply static/panes/tool.js:227]: considering compilerId's (pane id's) are unique, it should be fine

**Comment by partouf** [comment]
seems this doesn't fix the "change language and get double compiles" issue :(

**Pull request merged**